wandb/sdk/wandb_init.py

```diff
@@ -1152,6 +1152,30 @@ def _monkeypatch_tensorboard() -> None:
     tb_module.patch()
 
 
+def try_create_root_dir(settings: Settings):
+    """Try to create the root directory specified in settings.
+
+    This function attempts to create the root directory if it doesn't exist.
+    If creation fails due to permissions or other OS errors.
+    If creation of the root directory fails, the root_dir will be set to the
+    system temp directory.
```

Comment:
review the wording here (the second sentence especially), and consider making it more concise